Add the issue checker to enterprise scripts folder

Kingdutch opened the pull request:
This script looks at the extra.patches entry in a composer.json file and parses patches in the form of "issuenr - human readable description" to retrieve the issue status and display it in a pretty table.

![screen shot 2017-08-09 at 15 53 10](https://user-images.githubusercontent.com/327697/29125909-431635de-7d1d-11e7-9ca1-c01d3e4b3144.png)

Kingdutch committed:
Add the issue checker to enterprise scripts folder

Kingdutch committed:
Add GitHub pull request patch support

This issue adds support for pull request statusses through the
GitHub API v3.

Kingdutch commented:
![screen shot 2017-08-21 at 14 27 22](https://user-images.githubusercontent.com/327697/29519240-e44041a8-867c-11e7-9258-f5f8278b1b13.png)

The improved version adds support for GitHub PRs

bramtenhove reviewed (changes requested):
The index.js file has some indentation issues (around line 10) and extra blank lines around line 23.

Kingdutch committed:
Make composer.json optional as argument

This is most often going to be run from the directory where the
composer.json file resides so it doens't make sense to have to
specify the file as it's always known.

Kingdutch commented:
I've fixed the whitespace issues, improved the usage help text and made the `composer.json` argument optional in the process.

Kingdutch committed:
Add release status for GitHub pull requests

This commit adds the bulk of logic and requests required to find
the first tag that was created after a pull request was merged.

Changes nothing for unmerged pull requests.

Kingdutch committed:
Add release reporting support for Drupal.org

Kingdutch committed:
Take 'authored on' type messages into account for drupal.org

Kingdutch committed:
Add a --help command

Kingdutch committed:
Add exists check for composer file

Kingdutch committed:
Update version and add author

bramtenhove commented on issue_checker/readme.md:
````diff
@@ -0,0 +1,10 @@
+## Usage
+
+```
+/path/to/issue_checker/index.js /path/to/composer.json
````
@Kingdutch the composer.json was not needed right? Can you update it in the readme?

Kingdutch replied:
You're right, I've fixed it

Kingdutch committed:
Update README for optional composer path
